packages/mobile/src/send/saga.ts

```diff
@@ -188,17 +200,124 @@ function* sendPayment(
       txId: context.id,
       recipientAddress,
       amount: amount.toString(),
-      currency,
+      tokenAddress: currency,
+      usdAmount: '',
     })
     yield call(giveProfileAccess, recipientAddress)
   } catch (error) {
-    Logger.error(`${TAG}/sendPayment`, 'Could not send payment', error)
+    Logger.error(`${TAG}/sendPaymentLegacy`, 'Could not send payment', error.message)
     ValoraAnalytics.track(SendEvents.send_tx_error, { error: error.message })
     throw error
   }
 }
 
-export function* sendPaymentOrInviteSaga({
+function* buildSendTx(
+  tokenAddress: string,
+  amount: BigNumber,
+  recipientAddress: string,
+  comment: string
+) {
+  const contract: Contract = yield call(getERC20TokenContract, tokenAddress)
+  const coreContract: CeloTokenWrapper<any> | undefined = yield call(
+    getTokenContractFromAddress,
+    tokenAddress
+  )
+  const convertedAmount: string = yield call(tokenAmountInWei, amount, tokenAddress)
+
+  const kit: ContractKit = yield call(getContractKit)
+  return coreContract
```

Comment:
What does it mean that `coreContract` is `undefined`?

Reply:
There is a `coreContract` for cUSD, cEUR and CELO. These core contracts have the `transferWithComment` function that the normal ERC20 interface doesn't.

So, what we do here is: if there is a core contract, call `transferWithComment` on it, otherwise (for all other tokens) just do a normal `transfer` without any comment.